### PR TITLE
fix: 9870 merge button visible condition

### LINKED ISSUE #9870: [Bug]: The merge changes button should be displayed when the user tries to merge changes once again after the first merge
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

In The merge tab when the user merge the changes to master from his branch2 and now to merge the same changes to branch 1  the merge Changes tab is not visible

### Steps To Reproduce

1. Connect your application to git
2. Now create. a branch 1 add the changes merge to master
3. Now from master create a new branch branch 2
4. do the changes and merge to master 
5. Now try to merge the changes to branch one without altering anything in the pop up
6. The merge changes button is not displayed

### Environment

Deploy Preview

### Version

Cloud

### PR DESCRIPTION
**Pull request by haojin111**
## Description

> updated conflicting condition of merge tab

Fixes #9870

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> refer attached issue

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/9870-merge-change-button-issue 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.12 **(0)** | 36.65 **(0.01)** | 34.54 **(0)** | 55.65 **(0)**
 :red_circle: | app/client/src/pages/Editor/gitSync/components/BranchList.tsx | 23.31 **(-0.29)** | 10.13 **(-0.54)** | 2.5 **(0)** | 25.34 **(-0.18)**
 :green_circle: | app/client/src/selectors/commentsSelectors.ts | 85.25 **(1.64)** | 64.71 **(2.95)** | 73.33 **(0)** | 90.59 **(2.35)**
 :red_circle: | app/client/src/selectors/gitSyncSelectors.tsx | 71.65 **(-0.12)** | 26.19 **(0)** | 16.67 **(-0.4)** | 64.1 **(0.46)**
 :red_circle: | app/client/src/utils/WorkerUtil.ts | 88.98 **(-0.78)** | 70.59 **(-1.96)** | 100 **(0)** | 92.38 **(-0.95)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**</details>

### CODE REVIEW

File: app/client/src/pages/Editor/gitSync/Tabs/Merge.tsx [hunk 1]
```diff
@@ -212,9 +215,14 @@ export default function Merge() {
...
     status = MERGE_STATUS_STATE.NOT_MERGEABLE;
+  } else if (mergeError) {
+    status = MERGE_STATUS_STATE.ERROR;
+    mergeStatusMessage = mergeError.error.message;
   }
 
-  const isConflicting = (mergeStatus?.conflictingFiles?.length || 0) > 0;
+  // should check after added error code for conflicting
+  const isConflicting =
+    !!mergeError || (mergeStatus && !mergeStatus?.isMergeAble);
```
Comment (rishabhsaxena) [inline app/client/src/pages/Editor/gitSync/Tabs/Merge.tsx:225]:
```suggestion
const isConflicting = (mergeStatus?.conflictingFiles?.length || 0) > 0;
```
We should keep this as existing for now, ideally it should take into consideration both the `mergeError` and `mergeStatus`

**Review by rishabhsaxena — COMMENTED** [review]
@haojin111 just a minor comment

File: app/client/src/pages/Editor/gitSync/Tabs/Merge.tsx (hunk 1 above)
Comment (haojin111) [reply app/client/src/pages/Editor/gitSync/Tabs/Merge.tsx]: good point

**Review by rishabhsaxena — APPROVED** [review]

**Pull request merged**